### DISCUSSION TITLE
Delete transfers and transactions

### DISCUSSION THREAD

**Discussion by juliaokataleko** [bavix/laravel-wallet#809] [Q&A]
Hello people. How to safetly delete the transafers and transactions?
When I delete using eloquent, the wallet doesn't change the value.

**Comment by rez1dent3** [comment]
Hello. I won't beat around the bush, there is no such functionality, but I once wanted to think about it. I won't promise implementation.

**Attention. I didn't check the deletion, I just wrote a thought. I do not recommend deleting transactions and transfers.**

I think:
1. Update entries via resetConfirm so that they are not taken into account in the balance;
2. Delete entries;

Example for laravel wallet 9.6+:
```php
use Illuminate\Support\Facades\DB;

DB::beginTransaction();

$user->resetConfirm($transactionForDelete);
$transactionForDelete->delete();

DB::commit();
```

https://bavix.github.io/laravel-wallet/#/confirm
https://bavix.github.io/laravel-wallet/#/transaction

https://github.com/bavix/laravel-wallet/blob/40f5dc1418a9073923107e6ba9defb05fa6c767d/src/Traits/CanConfirm.php#L72C21-L72C33

**Reply by rez1dent3** [reply]
**Attention. I didn't check the deletion, I just wrote a thought. I do not recommend deleting transactions and transfers.**

Transfer. Example for laravel wallet 9.6+:
```php
 use Illuminate\Support\Facades\DB;

DB::beginTransaction();

$transfer->withdraw->wallet->resetConfirm($transfer->withdraw);
$transfer->withdraw->delete();

$transfer->deposit->wallet->resetConfirm($transfer->deposit);
$transfer->deposit->delete();

$transfer->delete();

DB::commit();
```

#746

**Reply by juliaokataleko** [reply]
Ok. Thanks for your feed!! I'll think a good way to implement it in my system.
Or just i will disable transactions deletion how you have recommended.

**Reply by rez1dent3** [reply]
@juliaokataleko It’s not a fact that it will be included in the release. Testing will show this.

I implement soft deletes for wallet models, you can see the implementation here:
https://github.com/bavix/laravel-wallet/pull/827/files
https://github.com/bavix/laravel-wallet/pull/828/files
